src/Symfony/Component/Yaml/CHANGELOG.md

```diff
@@ -6,6 +6,20 @@ CHANGELOG
 
  * Added `github` format support & autodetection to render errors as annotations
    when running the YAML linter command in a Github Action environment.
+ * Added support for optional YAML lint config from file `lint:yaml --config=.yamllint.yml`.
+
+   ```yml
+   yaml-lint:
```

Comment:
I am not really sure that we need this top-level entry. IMO this would work as well:

```yaml
format: txt
includes:
  - src/Resources/config/foo.yml
  - src/Resources/config/bar.yml
  - src/Resources/config/path/
excludes:
  - src/Resources/config/path/exclude.yml
```

Reply:
Yes it works fine without the top level entry. The only technical reason for the tag is to check if the file has a "valid" lint config:

```php
if (!isset($result['yaml-lint'])) {
    throw new RuntimeException(sprintf('Invalid YAML lint config "%s".', $config));
}
```

And with the tag it is possible to mix the config with other configurations in the same file, not sure if someone need this but it is possible. 

Last but not least, the tag could help other developers to detect the reason for config file...

For me both is OK. What do you think?

Reply:
I think that it will be easier to write a config from scratch if we ditch the top-level `yaml-lint` entry. I would then do a naiv validation which will do a check like if `array_diff(array_keys($config), ['format', 'includes', 'excludes', 'parse-tags'])` returns the empty array.

Reply:
OK so, I have dropped the top level entry and changed the tests.